src/components/NewContactPage.jsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+function NewContactPage({ contacts, setContacts }) {
+  const [newContact, setNewContact] = useState({
+    firstName: "",
+    lastName: "",
+    street: "",
+    city: "",
+  });
+
+  const navigate = useNavigate();
+
+  const handleFormUpdate = (event) => {
+    //Load all data from this input
```

Comment:
This function can be simplified by using the id or name attribute of the input fields directly, rather than checking each field with an if statement